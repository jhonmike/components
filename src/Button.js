import classnames from 'classnames'
import PropTypes from 'prop-types'

import styles from './button.css'

const prepareProps = props => {
  const { className, theme, ...otherProps } = props
  const classes = classnames(
    styles.btn,
    theme ? styles[`btn-${theme}`] : null,
    className
  )

  return {
    ...otherProps,
    className: classes,
  }
}

const Button = props => {
  const { className, theme, children, ...otherProps } = prepareProps(props)

  return (
    <button {...otherProps} className={className}>
      {children}
    </button>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  theme: PropTypes.oneOf(['primary', 'secondary']),
  children: PropTypes.node,
}

export default Button
