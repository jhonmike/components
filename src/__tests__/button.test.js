import React from 'react'
import Button from '../Button'
import renderer from 'react-test-renderer'

describe('Button Component', () => {
  it('renders default component', () => {
    const tree = renderer.create(<Button>Button Default</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders primary component', () => {
    const tree = renderer
      .create(<Button theme="primary">Button Primary</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders secondary component', () => {
    const tree = renderer
      .create(<Button theme="secondary">Button Secondary</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
