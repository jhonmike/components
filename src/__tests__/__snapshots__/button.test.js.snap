// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders default component 1`] = `
<button
  className="btn btn-default"
>
  Button Default
</button>
`;

exports[`Button Component renders primary component 1`] = `
<button
  className="btn btn-primary"
>
  Button Primary
</button>
`;

exports[`Button Component renders secondary component 1`] = `
<button
  className="btn btn-secondary"
>
  Button Secondary
</button>
`;
